src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -176,6 +176,12 @@ object ResourceMatcher extends StrictLogging {
 
     def portsMatchOpt: Option[PortsMatch] = PortsMatcher(runSpec, offer, selector).portsMatch
 
+    val meetsFaultDomainRequirements: Boolean = {
+      val faultDomainFields = Set(Constraints.regionField, Constraints.zoneField)
+      val hasFaultDomainConstraints = runSpec.constraints.exists(c => faultDomainFields.contains(c.getField))
+      hasFaultDomainConstraints || homeFaultDomain.map(_.region) == OfferUtil.region(offer)
```

Comment:
Hm, I'm not such a big fan of comparing options. How about

```scala
homeFaultDomain.exists( domain => OfferUtil.region(offer).contains(domain.region) )
```

I don't have a strong opinion though.